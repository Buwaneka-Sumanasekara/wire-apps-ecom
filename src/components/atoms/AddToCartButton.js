import React from 'react';
import { Ionicons, MaterialCommunityIcons } from '@expo/vector-icons';
import { Text, TouchableOpacity, View } from 'react-native';

const AddToCartButton = ({ isMini, onPress, isDisabled }) => {
    return (
        <TouchableOpacity onPress={onPress} disabled={isDisabled}>
            {isMini ? <MaterialCommunityIcons name="cart-heart" size={25} color={(isDisabled ? "grey" : "black")} className={"self-end"} /> :
                <View className={"bg-primary flex flex-row  justify-center  items-center p-2 rounded-md"}>
                    <Ionicons name="heart" color={(isDisabled ? "grey" : "white")} size={25} />
                    <Text className={`${isDisabled ? 'text-slate-500' : 'text-white'} ml-1`}>{"Add To Cart"}</Text>
                </View>}
        </TouchableOpacity>
    )

}

export default AddToCartButton;