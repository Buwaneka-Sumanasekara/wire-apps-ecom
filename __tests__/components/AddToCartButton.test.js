import React from 'react';
import renderer from 'react-test-renderer';

import AddToCartButton from '../../src/components/atoms/AddToCartButton';




describe('<AddToCartButton />', () => {
    it('rendered sucessfully', () => {
        const tree = renderer.create(<AddToCartButton />).toJSON()
        expect(tree).toMatchSnapshot()
    });
   

});

