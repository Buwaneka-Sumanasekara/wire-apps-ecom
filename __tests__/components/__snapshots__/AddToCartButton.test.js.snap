// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddToCartButton /> rendered sucessfully 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#030303",
        },
        {
          "display": "flex",
        },
        {
          "flexDirection": "row",
        },
        {
          "justifyContent": "center",
        },
        {
          "alignItems": "center",
        },
        {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
        {
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderTopLeftRadius": 6,
          "borderTopRightRadius": 6,
        },
      ]
    }
  >
    <Text />
    <Text
      style={
        [
          {
            "color": "#fff",
          },
          {
            "marginLeft": 4,
          },
        ]
      }
    >
      Add To Cart
    </Text>
  </View>
</View>
`;
